src/main/java/com/maria/library/facade/implementation/BookFacade.java

```diff
@@ -49,4 +58,130 @@ public Response<BookDto> addBook(BookDto bookDto) {
+        List<Book> books = bookService.getAll();
+        List<BookAuthor> bookAuthors = bookAuthorService.getAll();
+        List<Author> authors = authorService.getAll();
+        List<Category> categories = categoryService.getAll();
+        List<BookDetailsDto> bookDetailsDtoList = new ArrayList<>();
+
+        bookAuthors = bookAuthors.stream().filter(ba -> ba.getAuthorId() == authorId).toList();
+
+        for (var bookAuthor : bookAuthors) {
+
+            var author = authors
+                    .stream()
+                    .filter(a -> a.getId() == bookAuthor.getAuthorId())
+                    .findFirst().get();
+
+            var book = books
+                    .stream()
+                    .filter(b -> b.getId() == bookAuthor.getBookId())
+                    .findFirst().get();
+
+            var category = categories
+                    .stream()
+                    .filter(c -> c.getId() == book.getCategoryId())
+                    .findFirst().get();
+
+            BookDetailsDto bookDetailsDto = bookMapper.mapToEntity(book, author, category);
+            bookDetailsDtoList.add(bookDetailsDto);
+        }
+        return new Response<>(bookDetailsDtoList);
+    }
+
+    @Override
+    public Response<List<BookDetailsDto>> getBooksByClient(Long clientId) {
+        List<Book> books = bookService.getAll();
+        List<BookAuthor> bookAuthors = bookAuthorService.getAll();
+        List<Author> authors = authorService.getAll();
+        List<Category> categories = categoryService.getAll();
+        List<BookDetailsDto> bookDetailsDtoList = new ArrayList<>();
+        books = books.stream().filter(b -> b.getClientId() == clientId).toList();
+
+        for (var book : books) {
+            var bookAuthor = bookAuthors
+                    .stream()
+                    .filter(b -> b.getBookId() == book.getId())
+                    .findFirst().get();
+
+            var author = authors
+                    .stream()
+                    .filter(a -> a.getId() == bookAuthor.getAuthorId())
+                    .findFirst().get();
+
+            var category = categories
+                    .stream()
+                    .filter(c -> c.getId() == book.getCategoryId())
+                    .findFirst().get();
+
+            BookDetailsDto bookDetailsDto = bookMapper.mapToEntity(book, author, category);
+            bookDetailsDtoList.add(bookDetailsDto);
+        }
+        return new Response<>(bookDetailsDtoList);
+    }
+
+    @Override
+    public Response<List<BookDetailsDto>> getAvailableBooks() {
+        List<Book> books = bookService.getAll();
+        List<BookAuthor> bookAuthors = bookAuthorService.getAll();
+        List<Author> authors = authorService.getAll();
+        List<Category> categories = categoryService.getAll();
+        List<BookDetailsDto> bookDetailsDtoList = new ArrayList<>();
+        books = books.stream().filter(b -> b.getClientId() == 0).toList();
+
+        for (var book : books) {
```

Comment:
you have this block of code multiple times. If possible, extract common code into a private method